File: Easy/217-Contains-Duplicate.rs
// https://leetcode.com/problems/contains-duplicate/submissions/959635161/

Runtime 16ms - Beats 81.29%
Memory 3.8 - Beats 62.65%

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        use std::collections::HashSet;
        let mut hash_set = HashSet::new();
        nums.iter().any(|&num| !hash_set.insert(num))
    }
}

File: Easy/217-Contains-Duplicate.js
// runtime beats 83.19%
// memory beats 68.42%

var containsDuplicate = function(nums) {
    var hset = new Set();
    for(var i = 0; i < nums.length; i++){
        // 1: check set to see it has the value
        if(hset.has(nums[i])){
            // 2: return true; duplicate was found
            return true;
        }
        // 3: add the value to the set
        hset.add(nums[i]);
    }
    // 4: duplicate wasn't found
    return false;
};
